src/reactionsystem.jl

```diff
@@ -1414,6 +1426,20 @@ function Base.convert(::Type{<:SDESystem}, rs::ReactionSystem;
               kwargs...)
 end
 
+# Extracts any noise scaling parameters from a reaction system.
+function get_noise_scaling(rs::ReactionSystem)
+    ns_params = filter(p -> is_noise_scaling_parameter(p), parameters(rs))
+    if isempty(ns_params)
+        return nothing
+    elseif length(ns_params) == 1
+        return Num(ns_params[1])
```

Comment:
Don't use `Num`s. I hope you can please try to keep this in mind going forward, as this seems to be a recurring issue.

Reply:
Sorry, will fix. Think I made a couple of these when I didn't really understand the difference between `Num` and other stuff. Maybe next time we meet you can go through the various symbolics representations, and why we use which in various places, understanding that would probably help.

Reply:
`Num` is a wrapper around all symbolics stuff to use in the REPL primarily. So a `Num` can wrap a single scalar symbolic, but also a symbolic expression. The rule in MTK/Catalyst is to always unwrap these via either `ModelingToolkit.unwrap(s::Num)` or `ModelingToolkit.value(s::Num)` to get the actual symbolic. There are some MTK/Symbolics functions that don't work with `Num`s, which will then cause issues when adding code if we store them. An example:
```julia
julia> using ModelingToolkit
@parameters k
julia> @parameters k
1-element Vector{Num}:
@v arik

julia> @variables t A(t)
2-element Vector{Num}:
    t
 A(t)

julia> using ModelingToolkit: value

julia> typeof(k),typeof(value(k))
(Num, SymbolicUtils.BasicSymbolic{Real})

julia> typeof(A),typeof(value(A))
(Num, SymbolicUtils.BasicSymbolic{Real})

julia> typeof(k*A),typeof(value(k*A))
(Num, SymbolicUtils.BasicSymbolic{Real})

julia> typeof(k*A),typeof(value(k*A^2))
(Num, SymbolicUtils.BasicSymbolic{Real})

julia> typeof(Num(1)),typeof(value(Num(1)))
(Num, Int64)

julia> typeof(k*A^2*sin(A)),typeof(value(k*A^2*sin(A)))
(Num, SymbolicUtils.BasicSymbolic{Real})
```
so it looks like everything is a `BasicSymbolic{Real}` now actually.